### PR TITLE
audit: Port audit_homepage method to rubocop and add tests

### PR DESCRIPTION
**Pull request by GauthamGoli**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/master/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [x] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/homebrew/pull/49031).
- [x] Have you successfully run `brew tests` with your changes locally?

-----
Ports all the rules defined in `audit_homepage` method except for the ones that execute with `--online` option , they will be placed in a separate cop category for online audit rules.

### CODE REVIEW

File: Library/Homebrew/rubocops/homepage_cop.rb [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+  module Cop
+    module FormulaAudit
+      # This cop audits `homepage` url in Formulae
+      class Homepage < FormulaCop
+        def audit_formula(_node, _class_node, _parent_class_node, formula_class_body_node)
+          homepage_node = find_node_method_by_name(formula_class_body_node, :homepage)
+          homepage = homepage_node.nil? ? "" : string_content(parameters(homepage_node).first)
+
+          problem "Formula should have a homepage." if homepage_node.nil? || homepage.empty?
+          problem "The homepage should start with http or https (URL is #{homepage})." unless homepage =~ %r{^https?://}
```
Comment (MikeMcQuaid) [inline Library/Homebrew/rubocops/homepage_cop.rb]: This line is pretty long so I'd stick the `unless` onto the previous line.

File: Library/Homebrew/rubocops/homepage_cop.rb [hunk 2]
```diff
@@ -0,0 +1,78 @@
+require_relative "./extend/formula_cop"
+
+module RuboCop
+  module Cop
+    module FormulaAudit
+      # This cop audits `homepage` url in Formulae
+      class Homepage < FormulaCop
+        def audit_formula(_node, _class_node, _parent_class_node, formula_class_body_node)
+          homepage_node = find_node_method_by_name(formula_class_body_node, :homepage)
+          homepage = homepage_node.nil? ? "" : string_content(parameters(homepage_node).first)
```
Comment (MikeMcQuaid) [inline Library/Homebrew/rubocops/homepage_cop.rb]: I'd do something like:
```ruby
homepage = if homepage_node
  string_content(parameters(homepage_node).first) 
else
  ""
end
```

File: Library/Homebrew/rubocops/homepage_cop.rb [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+
+          problem "Formula should have a homepage." if homepage_node.nil? || homepage.empty?
+          problem "The homepage should start with http or https (URL is #{homepage})." unless homepage =~ %r{^https?://}
+
+          # Check for http:// GitHub homepage urls, https:// is preferred.
+          # Note: only check homepages that are repo pages, not *.github.com hosts
+          problem "Please use https:// for #{homepage}" if homepage.start_with? "http://github.com/"
```
Comment (MikeMcQuaid) [inline Library/Homebrew/rubocops/homepage_cop.rb]: I wonder if it's worth turning this and everything below into a big `case` statement? What do you think?

Comment (GauthamGoli) [inline Library/Homebrew/rubocops/homepage_cop.rb]: Yeah, I think its better that way.

**Comment by MikeMcQuaid** [comment]
Great work again @GauthamGoli!

**Comment by GauthamGoli** [comment]
Thanks for the review! :)

**Pull request merged**